packages/client/src/request-login.test.ts

```diff
@@ -1,11 +1,110 @@
+import { DEFAULT_SCOPE_STRING } from './constants';
 import { getLoginUrlAndCodeVerifier } from './request-login';
 
-test('getLoginUrlAndCodeVerifier', async () => {
-  const { url } = await getLoginUrlAndCodeVerifier({
-    baseUrl: 'http://logto.dev/oidc/auth',
-    clientId: 'foo',
-    scope: 'openid offline_access',
-    redirectUri: 'http://localhost:3000',
+const BASE_URL = 'http://logto.dev/oidc/auth';
+const CLIENT_ID = 'foo';
+const REDIRECT_URI = 'http://localhost:3000';
+
+describe('getLoginUrlAndCodeVerifier', () => {
+  describe('url must contain required parameters', () => {
```

Comment:
why do we need to wrap up two `describe` methods? feels like all test case are running under a same testing context

Reply:
also you could define the url at the description level as they all shares a same input for every test case.

Reply:
fixed.